fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1292,6 +1292,11 @@ ACTOR Future<Void> commitBatch(
 		ASSERT(p.second.isReady());
 	}
 
+	if (SERVER_KNOBS->ASK_READ_VERSION_FROM_MASTER) {
```

Comment:
figure out why this cannot be inside of if( commitVersion > self->committedVersion.get() )

Reply:
The reason is under my newly added lines:
if (commitVersion > self->committedVersion.get())
	self->committedVersion.set(commitVersion);
After a proxy reports a commit version v1 to the master and when it's waiting, the event loop will run other commitBatch events which may have finished reporting commit version and finished setting the committedVersion to a larger commit version v2 that is larger than v1. After that we will fail an assertion [1] when setting a NotifiableVersion of v2 to be v1 and v2 > v1. The newly added if statement will fix this.

[1] https://github.com/apple/foundationdb/blob/99bf993815d0363862b870038d2c94fb79cd293d/fdbclient/Notified.h#L58